packages/perseus-core/src/utils/util.ts

```diff
@@ -0,0 +1,688 @@
+        // @ts-expect-error - TS2769 - No overload matches this call.
+        window.removeEventListener("testPassive", null, opts);
+    } catch {
+        // Intentionally left empty!
+    }
+
+    return supportsPassive;
+};
+
+/**
+ * Gets the word right before where the textarea cursor is
+ *
+ * @param {Element} textarea - The textarea DOM element
+ * @return {JSON} - An object with the word and its starting and ending positions in the textarea
+ */
+function getWordBeforeCursor(textarea: HTMLTextAreaElement): WordAndPosition {
+    const text = textarea.value;
+
+    const endPos = textarea.selectionStart - 1;
+    const startPos =
+        Math.max(
+            text.lastIndexOf("\n", endPos),
+            text.lastIndexOf(" ", endPos),
+        ) + 1;
+
+    return {
+        string: text.substring(startPos, endPos + 1),
+        pos: {
+            start: startPos,
+            end: endPos,
+        },
+    };
+}
+
+/**
+ * Moves the textarea cursor at the specified position
+ *
+ * @param {Element} textarea - The textarea DOM element
+ * @param {int} pos - The position where the cursor will be moved
+ */
+function moveCursor(textarea: HTMLTextAreaElement, pos: number): void {
+    textarea.selectionStart = pos;
+    textarea.selectionEnd = pos;
+}
+
+const textarea = {
+    getWordBeforeCursor,
+    moveCursor,
+} as const;
+
+/**
+ * Many of our labels are automatically converted into math mode without
+ * the dollar signs. Unfortunately, this makes them untranslatable! This
+ * helper function removes the math mode symbols from a string if we want
+ * to translate it but don't need the extra dollar signs.
+ */
+const unescapeMathMode: (label: string) => string = (label) =>
+    label.startsWith("$") && label.endsWith("$") ? label.slice(1, -1) : label;
+
+const random: RNG = seededRNG(new Date().getTime() & 0xffffffff);
+
+// TODO(benchristel): in the future, we may want to make deepClone work for
+// Record<string, Cloneable> as well. Currently, it only does arrays.
+type Cloneable =
+    | null
+    | undefined
+    | boolean
+    | string
+    | number
+    | Cloneable[]
+    | readonly Cloneable[];
+function deepClone<T extends Cloneable>(obj: T): T {
+    if (Array.isArray(obj)) {
+        return obj.map(deepClone) as T;
+    }
+    return obj;
+}
+
+const Util = {
+    inputPathsEqual,
+    nestedMap,
+    rWidgetRule,
+    rTypeFromWidgetId,
+    rWidgetParts,
+    snowman,
+    seededRNG,
+    shuffle,
+    split,
+    stringArrayOfSize,
+    gridDimensionConfig,
+    getGridStep,
+    snapStepFromGridStep,
+    scaleFromExtent,
+    tickStepFromExtent,
+    gridStepFromTickStep,
+    tickStepFromNumTicks,
+    constrainedTickStepsFromTickSteps,
+    eq,
+    deepEq,
+    parseQueryString,
+    updateQueryString,
+    strongEncodeURIComponent,
+    touchHandlers,
+    resetTouchHandlers,
+    extractPointerLocation,
+    supportsPassiveEvents,
+    textarea,
+    unescapeMathMode,
+    random,
+    deepClone,
+} as const;
```

Comment:
What's your appetite like for switching these to simple named exports instead of bundling them in an object first? We could still export them as a `Util` namespace in the top-level `index.ts` so the imports don't change...